contracts/multisig/test/asset_proxy_owner.ts

```diff
@@ -1,511 +1,641 @@
+                data,
+                destinations,
+                signerAddresses,
+                DEFAULT_TIME_LOCK,
+                { values },
+            );
+            assertReceiverCalledFromLogs(results.executionTxReceipt.logs, data, destinations, results.txId, values);
+        });
+        it('should be able to call a combination of registered and unregistered functions', async () => {
+            const data = [hexRandom(), hexRandom()];
+            const destinations = [receiver.address, receiver.address];
+            const newTimeLock = new BigNumber(DEFAULT_TIME_LOCK).dividedToIntegerBy(2);
+            await assetProxyOwner.registerFunctionCallBypassWallet.awaitTransactionSuccessAsync(
+                true,
+                data[0].slice(0, 10),
+                receiver.address,
+                newTimeLock,
+            );
+            const results = await assetProxyOwnerWrapper.submitConfirmAndExecuteTransactionAsync(
+                data,
+                destinations,
+                signerAddresses,
+                DEFAULT_TIME_LOCK,
+            );
+            assertReceiverCalledFromLogs(results.executionTxReceipt.logs, data, destinations, results.txId);
+        });
+        it('should fail if a single function has not passed the timelock', async () => {
+            const data = [hexRandom(), hexRandom()];
+            const destinations = [receiver.address, receiver.address];
+            const newTimeLock = new BigNumber(DEFAULT_TIME_LOCK).dividedToIntegerBy(2);
+            await assetProxyOwner.registerFunctionCallBypassWallet.awaitTransactionSuccessAsync(
+                true,
+                data[0].slice(0, 10),
+                receiver.address,
+                newTimeLock,
+            );
+            const tx = assetProxyOwnerWrapper.submitConfirmAndExecuteTransactionAsync(
+                data,
+                destinations,
+                signerAddresses,
+                newTimeLock.toNumber(),
+            );
+            expect(tx).to.revertWith(RevertReason.DefaultTimeLockIncomplete);
+        });
+        it('should be able to execute a transaction if called by any address', async () => {
+            const data = [hexRandom()];
+            const destinations = [receiver.address];
+            const results = await assetProxyOwnerWrapper.submitConfirmAndExecuteTransactionAsync(
+                data,
+                destinations,
+                signerAddresses,
+                DEFAULT_TIME_LOCK,
+                { executeFromAddress: notSignerAddress },
+            );
+            assertReceiverCalledFromLogs(results.executionTxReceipt.logs, data, destinations, results.txId);
+        });
+        it('should be able to send value without data if past the default timelock', async () => {
+            // NOTE: elements of `data` must be at least 4 bytes long
+            const data = [constants.NULL_BYTES4];
```

Comment:
It would probably be better if zero data was permissible. Would this be a hard change to make? I feel like it's a bit weird to have to send data in this kind of a scenario.

Reply:
We'd just need to special case it. Since I don't think this contract will ever actually be used to send ETH, I decided to leave it as is.